Singer.API/Controllers/CareUserController.cs

```diff
@@ -1,114 +1,47 @@
 using System;
 using System.Threading.Tasks;
-using AutoMapper;
-using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
 using Singer.DTOs.Users;
-using Singer.DTOs;
 using Singer.Models.Users;
-using Singer.Services;
+using Singer.Helpers.Exceptions;
+using Singer.Services.Interfaces;
 
 namespace Singer.Controllers
 {
    [Route("api/[controller]")]
-   [Authorize()]
-   public class CareUserController : Controller
+   //[Authorize()]
+   public class CareUserController : DataControllerBase<CareUser, CareUserDTO, CreateCareUserDTO, UpdateCareUserDTO>
    {
-      private readonly CareUserService _userService;
-      private readonly IMapper _mapper;
-
-      public CareUserController(CareUserService service, IMapper mapper)
+      private readonly ICareUserService careUserService;
+      public CareUserController(ICareUserService databaseService) : base(databaseService)
```

Comment:
I'd expect to see a `careUserService` instead of the `databaseService` variable.